src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import classnames from 'classnames';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  setMovies: (value: any) => void,
+};
+
+
+export const FindMovie: React.FC<Props> = ({ setMovies }) => {
+  const [query, setQuery] = useState('');
+  const [search, setSearch] = useState<Movie | undefined>();
+  const [isFailed, setIsFailed] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const defaultPicture
+    = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+  const addMovieOnClick = () => {
+    setMovies((prevArray: Movie[]) => {
+      const uniqMovieStatus = [...prevArray].some(
+        item => item.imdbId === search?.imdbId,
+      );
+
+      if (uniqMovieStatus) {
+        return prevArray;
+      }
+
+      return [...prevArray, search];
+    });
+
+    setSearch(undefined);
+    setIsLoading(false);
+    setIsFailed(false);
+  };
+
+  const loadMovie = async () => {
+    setIsLoading(true);
+    const movie = await getMovie(query);
+
+    if ('Error' in movie) {
+      setIsFailed(true);
+    } else {
+      setSearch({
+        title: movie.Title,
+        description: movie.Plot,
+        imgUrl: movie.Poster === 'N/A' ? defaultPicture : movie.Poster,
+        imdbUrl: `https://www.imdb.com/title/${movie.imdbID}`,
+        imdbId: movie.imdbID,
+      });
+
+      setIsFailed(false);
+    }
+
+    setQuery('');
+  };
 
-export const FindMovie: React.FC = () => {
   return (
     <>
-      <form className="find-movie">
+      <form
+        className="find-movie"
+        onSubmit={(event) => {
+          event.preventDefault();
+          loadMovie();
+        }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)